contracts/AMM/ThalesAMM.sol

```diff
@@ -442,34 +416,36 @@ contract ThalesAMM is Initializable, ProxyOwned, ProxyPausable, ProxyReentrancyG
 
         address target = _getTarget(market, position);
 
+        address _liquidityPool = liquidityPool.getOrCreateMarketPool(market);
+
         //transfer options first to have max burn available
-        IERC20Upgradeable(target).safeTransferFrom(msg.sender, address(this), amount);
+        IERC20Upgradeable(target).safeTransferFrom(msg.sender, _liquidityPool, amount);
 
         uint sUSDFromBurning = IPositionalMarketManager(manager).transformCollateral(
-            IPositionalMarket(market).getMaximumBurnable(address(this))
+            IPositionalMarket(market).getMaximumBurnable(_liquidityPool)
         );
         if (sUSDFromBurning > 0) {
-            IPositionalMarket(market).burnOptionsMaximum();
+            IPositionalMarket(market).burnOptionsMaximum(_liquidityPool);
+        }
+
+        uint safeBoxShare = (pricePaid * ONE) / (ONE - (safeBoxImpact)) - (pricePaid);
+        if (safeBoxImpact == 0) {
+            safeBoxShare = 0;
         }
 
+        liquidityPool.commitTrade(market, pricePaid + safeBoxShare);
```

Comment:
This needs to send the sUSD that the AMM needs to send to the user, but it can exclude the sUSD the AMM got from burning.